notebooks/tutorials/cloud-ml-engine/Model training and prediction with Cloud Machine Learning Engine.ipynb

```diff
@@ -0,0 +1,571 @@
+    "                    ('SKB-{}'.format(i), skb),\n",
+    "                    ('LBN-{}'.format(i), lbn)])\n",
+    "            )\n",
+    "        )\n",
+    "\n",
+    "# Create pipeline to extract the numerical features\n",
+    "skb = SelectKBest(k=6)\n",
+    "# From COLUMNS use the features that are numerical\n",
+    "skb.scores_ = [1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 0]\n",
+    "categorical_pipelines.append((\"numerical\", skb))\n",
+    "\n",
+    "# Combine all the features using FeatureUnion\n",
+    "preprocess = FeatureUnion(categorical_pipelines)\n",
+    "\n",
+    "# Create the classifier\n",
+    "classifier = RandomForestClassifier()\n",
+    "\n",
+    "# Transform the features and fit them to the classifier\n",
+    "classifier.fit(preprocess.transform(train_features), train_labels)\n",
+    "\n",
+    "# Create the overall model as a single pipeline\n",
+    "pipeline = Pipeline([(\"union\", preprocess), (\"classifier\", classifier)])\n",
+    "\n",
+    "# Create the model file\n",
+    "# It is required to name the model file \"model.pkl\" if you are using pickle\n",
+    "model_filename = \"model.pkl\"\n",
+    "with open(model_filename, \"wb\") as model_file:\n",
+    "    pickle.dump(pipeline, model_file)\n",
+    "\n",
+    "# Upload the model to Cloud Storage\n",
+    "bucket = storage_client.bucket(bucket_name)\n",
+    "blob = bucket.blob(model_filename)\n",
+    "blob.upload_from_filename(model_filename)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Submit the training job\n",
+    "\n",
+    "In this section, you use `gcloud ml-engine jobs submit training` to submit your training job. The `--` argument passed to the command is a separator; anything after the separator will be passed to the Python code as input arguments.\n",
+    "\n",
+    "For more information about the arguments preceeding the separator, run the following:\n",
+    "\n",
+    "        gcloud ml-engine jobs submit training --help\n",
+    "\n",
+    "The argument given to the python script is `--bucket-name`. The `--bucket-name` argument is used to specify the name of the bucket to save the model file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "\n",
+    "# Define a timestamped job name\n",
+    "JOB_NAME = \"census_training_{}\".format(int(time.time()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Submit the training job:\n",
+    "!gcloud ml-engine jobs submit training $JOB_NAME \\\n",
+    "  --job-dir gs://$BUCKET_NAME/census_job_dir \\\n",
+    "  --package-path ./census_training \\\n",
+    "  --module-name census_training.train \\\n",
+    "  --region us-central1 \\\n",
+    "  --runtime-version=1.9 \\\n",
```

Comment:
@happyhuman tested it with runtime version 1.12 and it appears to work.

Reply:
Updated version